### PR TITLE
Add set docker registry document

### PR DESCRIPTION
**Pull request by xuweibj**
The document is about how to set up a docker registry.
http://10.3.5.4/install/html/advanced/docker/docker_registry.html

### CODE REVIEW

File: docs/source/advanced/docker/docker_registry.rst [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+=======================
+
+Docker Registry is a stateless, high scalable server side application that stores and lets you distribute Docker images.
+
+In this document, it describes how to set a local private docker registry.
+
+**Note:** The Registry is compatible with Docker engine version 1.6.0 or higher.
+
+Prerequisite: Install Docker Compose
+----------------------
```
Comment (whowutwut) [inline docs/source/advanced/docker/docker_registry.rst]: This line is too short, must match text above. 

I don't think we should write "prerequisite". If this step is required then just write it in the text below. The heading can be "Install Docker Compose"

"Docker compose is needed... Etc.."

File: docs/source/advanced/docker/docker_registry.rst [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+Docker has an unusual mechanism for specifying which registry to push to. You have to tag an image with the private registry's location in order to push to it. Let's tag our image to our private registry ::
+
+    docker tag imagename domainname:5000/imagename
+
+Now we can push that image to our registry. ::
+
+    docker push domainname:5000/imagename
+
+**note:** Maybe there is some problem with the CA certificate. Edit the file ``/etc/default/docker`` so that there is a line that reads: ``DOCKER_OPTS="--insecure-registry domianname:5000"`` . Then restart Docker daemon ``service docker restart`` .
```
Comment (whowutwut) [inline docs/source/advanced/docker/docker_registry.rst]: Suggest: if there is a problem with the CA certificate, edit the file..

**Comment by gurevichmark** [comment]
I am working on setting up an environment to recreate described steps.

But my first comment is to add to the document a statement that these instructions are for x86_64 architecture. I do not think there are docker host or docker compose packages that a user can easily find for ppc64 architecture.

**Comment by daniceexi** [comment]
Yes, the setup steps just were verified for setting up docker regestry on x86 arch. But it should work for ppc64 too even there is no images for ppc so far.

BTW, docker registry on x86 supports ppc64 images.

**Comment by gurevichmark** [comment]
Few additional questions:
1. Should docker registry be set up on the docker host machine, or can the docker registry run on the system which is not a docker host ?
2. In the section "Create Configuration File", the instructions say "Define configuration file docker-compose.yml as below.", but it does not say in which directory. Is it implied that the directory is "/docker-registry" from the previous step?
3. In the section "Starting Docker Registry as a Service", I created docker-registry.service file in /etc/systemd/system/, and then added the contents shown. However, when I issued command "service docker-registry start", I got the following message back "docker-registry: unrecognized service". My system is c910f04x37 running Ubuntu 14.04.3

**Comment by xuweibj** [comment]
Hi @gurevichmark , 
Question 1: Docker registry must be set up on the docker host, and for our document, docker engine's version must be 1.6.0 or higher.
Question 2: Yes. The file's path is /docker-registry/docker-compose.yml
Question 3: Sorry, I did not note the system information I used. I created my registry on Ubuntu 15.04 on x86 arch, and Ubuntu 14.04 uses Upstart as the init system, not systemd. So for 14.04, it seems that must use Upstart to create service.

**Comment by daniceexi** [comment]
@xuweibj @gurevichmark Base the limitation of certificates, except the docker registry needs be on a docker host, it also needs be set up on the xCAT MN so far.

**Comment by gurevichmark** [comment]
In the section "Accessing Your Docker Registry from a Client Machine", the following command is listed:

**scp username@xCAT_MN_ip:/etc/xcat/cert/ca.pem /etc/docker/certs.d/domainname:5000/ca.crt**

did you really mean to have a *domainname:5000* as the directory name when copying the file ?

File: docs/source/advanced/docker/docker_registry.rst [hunk 3]
```diff
@@ -0,0 +1,173 @@
+Docker Registry in xCAT
+=======================
+
+Docker Registry is a stateless, high scalable server side application that stores and lets you distribute Docker images.
```
Comment (gurevichmark) [inline docs/source/advanced/docker/docker_registry.rst]: Change "high" to "highly"

File: docs/source/advanced/docker/docker_registry.rst [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+
+In this document, it describes how to set up a local private docker registry on Ubuntu 15.04 on x86_64.
```
Comment (gurevichmark) [inline docs/source/advanced/docker/docker_registry.rst]: Change to: "This document describes how to set up....."

File: docs/source/advanced/docker/docker_registry.rst [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+
+**Note:** Please ensure there is no docker registry had set up on this docker host.
```
Comment (gurevichmark) [inline docs/source/advanced/docker/docker_registry.rst]: Change to ".... Please ensure that docker registry in not already set up on ....."

File: docs/source/advanced/docker/docker_registry.rst [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+
+Setting Up Docker Host
+----------------------
+
+Install Docker version 1.6.0 or newer.
+
+Setting Up Docker Registry Manually
+-----------------------------------
+
+Now, docker registry needed to be set up on xCAT's MN.
```
Comment (gurevichmark) [inline docs/source/advanced/docker/docker_registry.rst]: No need for "Now".

File: docs/source/advanced/docker/docker_registry.rst [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+
+In this document, it describes two methods to set up docker registry manullay.
```
Comment (gurevichmark) [inline docs/source/advanced/docker/docker_registry.rst]: Change to: "This section describes two methods of setting up docker registry manually."

File: docs/source/advanced/docker/docker_registry.rst [hunk 8]
```diff
@@ -0,0 +1,173 @@
...
+
+At first, create a folder where our files for this tutorial will live and some of the subfolders we'll need. ::
```
Comment (gurevichmark) [inline docs/source/advanced/docker/docker_registry.rst]: Change to: "First, create some folders where files for this tutorial will live."

File: docs/source/advanced/docker/docker_registry.rst [hunk 9]
```diff
@@ -0,0 +1,173 @@
...
+
+Compose can also be run inside a container, from a small bash script wrapper. To install compose as a container run: ::
+
+    curl -L https://github.com/docker/compose/releases/download/1.5.2/docker-compose-`uname -s`-`uname -m` > /usr/local/bin/docker-compose
+    chmod +x /usr/local/bin/docker-compose
+
+Create Configuration File
+'''''''''''''''''''''''''
+
+Define configuration file ``docker-compose.yml`` as below. ::
```
Comment (gurevichmark) [inline docs/source/advanced/docker/docker_registry.rst]: Recommend adding which folder this file needs to be in.

File: docs/source/advanced/docker/docker_registry.rst [hunk 10]
```diff
@@ -0,0 +1,173 @@
...
+    WantedBy=default.target
+
+Start registry service: ::
+
+    service docker-registry start
+
+Accessing Docker Registry from other docker host
+------------------------------------------------
+
+Copy xCAT CA to docker host which is a client machine ::
```
Comment (gurevichmark) [inline docs/source/advanced/docker/docker_registry.rst]: Change to: "Copy ca.crt file from xCAT MN to a client machine. Client machine must be a docker host."

File: docs/source/advanced/docker/docker_registry.rst [hunk 11]
```diff
@@ -0,0 +1,173 @@
...
+
+    scp username@xCAT_MN_ip:/etc/xcat/cert/ca.pem /etc/docker/certs.d/domainname:5000/ca.crt
+
+Now you can push/pull images to/from the registry.
```
Comment (gurevichmark) [inline docs/source/advanced/docker/docker_registry.rst]: I think this line can be deleted.

File: docs/source/advanced/docker/docker_registry.rst [hunk 12]
```diff
@@ -0,0 +1,173 @@
...
+Pull Images from Registry
+`````````````````````````  
+Just use the "tag" image name, which includes the domain name, port, and image name. ::
+
+    docker pull domainname:5000/imagename
+
+Push Images to Registry
+```````````````````````
+
+Docker has an unusual mechanism for specifying which registry to push to. You have to tag an image with the private registry's location in order to push to it. Let's tag our image to our private registry: ::
```
Comment (gurevichmark) [inline docs/source/advanced/docker/docker_registry.rst]: Change to:
"Before the image can be pushed to the registry, it must be tagged with the location of the private registry:"

**Comment by xuweibj** [comment]
Hi @gurevichmark ,
Thank you for your comments, I modified as your comments, please help review again.

**Comment by gurevichmark** [comment]
Changes look good to me.

**Pull request merged**